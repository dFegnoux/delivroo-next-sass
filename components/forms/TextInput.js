import React from "react";

const TextInput = ({
  field, // { name, value, onChange, onBlur }
  form: { touched, errors }, // also values, setXXXX, handleXXXX, dirty, isValid, status, etc.
  label,
  className,
  ...props
}) => (
  <div className={`input col-${className}`}>
    <label>{label}</label>
    <input type="text" {...field} {...props} />
    {touched[field.name] && errors[field.name] && (
      <div className="error">{errors[field.name]}</div>
    )}
  </div>
);

export default TextInput;
