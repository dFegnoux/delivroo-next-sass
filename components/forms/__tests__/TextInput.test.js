import React from "react";
import renderer from "react-test-renderer";
import TextInput from "../TextInput";

const props = {
  className: "mockdeClassName",
  label: "mockedLabel",
  field: {
    name: "mockedFieldName"
  },
  form: {
    touched: {}
  }
};

describe("TextInput snapshots", () => {
  it("should render correctly with default props", () => {
    const tree = renderer.create(<TextInput {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should handle display of an error message", () => {
    const errorProps = {
      form: {
        errors: {
          [props.field.name]: "error"
        },
        touched: {
          [props.field.name]: true
        }
      }
    };
    const tree = renderer
      .create(<TextInput {...props} {...errorProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
