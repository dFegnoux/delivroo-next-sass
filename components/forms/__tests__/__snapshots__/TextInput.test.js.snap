// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput snapshots should handle display of an error message 1`] = `
<div
  className="input col-mockdeClassName"
>
  <label>
    mockedLabel
  </label>
  <input
    name="mockedFieldName"
    type="text"
  />
  <div
    className="error"
  >
    error
  </div>
</div>
`;

exports[`TextInput snapshots should render correctly with default props 1`] = `
<div
  className="input col-mockdeClassName"
>
  <label>
    mockedLabel
  </label>
  <input
    name="mockedFieldName"
    type="text"
  />
</div>
`;
